File: src/main/java/BB/resto/Entity/Burger.java
package BB.resto.Entity;

import BB.resto.Entity.Enumeration.TypeBurger;
import BB.resto.Entity.Enumeration.TypeProduct;

public class Burger extends Product {
    private TypeBurger typeBurger;
    private int calories;
    private int tempsPreparation;


    public Burger() {
        super();
    }

    public Burger(String libelle, String description, double prix, TypeBurger typeBurger, int calories, int tempsPreparation) {
        super(libelle, description, prix, TypeProduct.BURGER);
        this.typeBurger = typeBurger;
        this.calories = calories;
        this.tempsPreparation = tempsPreparation;
    }


    public TypeBurger getTypeBurger() { return typeBurger; }
    public void setTypeBurger(TypeBurger typeBurger) { this.typeBurger = typeBurger; }

    public int getCalories() { return calories; }
    public void setCalories(int calories) { this.calories = calories; }

    public int getTempsPreparation() { return tempsPreparation; }
    public void setTempsPreparation(int tempsPreparation) { this.tempsPreparation = tempsPreparation; }

    @Override
    public String toString() {
        return "Burger{" +
                "id=" + getId() +
                ", libelle='" + getLibelle() + '\'' +
                ", typeBurger=" + typeBurger +
                ", calories=" + calories +
                ", tempsPreparation=" + tempsPreparation + "min" +
                ", prix=" + getPrix() +
                '}';
    }
}

File: BBurgerSymfony/BBurgerSymfony/src/Entity/Burger.php
<?php
namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity]
#[ORM\Table(name: 'burger')]
class Burger extends Product
{
    #[ORM\Column(length: 20, nullable: true)]
    private ?string $typeBurger = null;

    #[ORM\Column(nullable: true)]
    private ?int $calories = null;

    #[ORM\Column(nullable: true)]
    private ?int $tempsPreparation = null;


    public function getTypeBurger(): ?string
    {
        return $this->typeBurger;
    }
    public function setTypeBurger(?string $typeBurger): self
    {
        $this->typeBurger = $typeBurger;
        return $this;
    }
    public function getCalories(): ?int
    {
        return $this->calories;
    }
    public function setCalories(?int $calories): self

    {
        $this->calories = $calories;
        return $this;
    }
    public function getTempsPreparation(): ?int
    {
        return $this->tempsPreparation;
    }
    public function setTempsPreparation(?int $tempsPreparation): self
    {
        $this->tempsPreparation = $tempsPreparation;
        return $this;
    }

    public function hasImage(): bool
    {
        return $this->getCloudinaryUrl() !== null && $this->getCloudinaryUrl() !== '';
    }

    public function getFormattedPrix(): string
    {
        return number_format((float) $this->getPrix(), 0, ',', ' ') . ' FCFA';
    }

}